diofant/integrals/tests/test_meijerint.py

```diff
@@ -651,3 +651,8 @@ def test_issue_6860():
 def test_issue_8368():
     assert meijerint_indefinite(cosh(x)*exp(-x*t), x) == (
         (-t - 1)*exp(x) + (-t + 1)*exp(-x))*exp(-t*x)/2/(t**2 - 1)
+
+
+def test_issue_4311():
+    assert meijerint_indefinite(x*abs(9 - x**2), x) == \
+        -81*meijerg(((3, 1), (2,)), ((1,), (2, 0)), x**2/9)
```

Comment:
Meanwhile I'll add a watchdog for nan in the meijerint_indefinite_1 and a simple heuristics to handle definite integrals with abs-like stuff (i.e. anything, that can be rewritten with Piecewiese's).  See https://github.com/diofant/diofant/pull/321.

I suggest you to revert your patch quickly to avoid new wrong answers.  Not sure if we can extend lockup_table with such limiting cases.  At least, it's not so simple.